src/shell/shell.cpp

```diff
@@ -736,6 +754,98 @@ class AUTOEXEC final : public Module_base {
 	}
 };
 
+// Specify a 'Drive' config object with allowed key and value types
+static Config specify_drive_config()
+{
+	auto conf = Config();
+
+	// Define the [drive] section
+	constexpr auto changeable_at_runtime = false;
+	auto prop = conf.AddSection_prop("drive", nullptr, changeable_at_runtime);
+
+	// Define the allowed keys and types
+	constexpr auto on_startup = Property::Changeable::OnlyAtStart;
+	const char *drive_types[] = {"dir", "floppy", "cdrom", "overlay", nullptr};
+	(void)prop->Add_string("type", on_startup, "")->Set_values(drive_types);
+	(void)prop->Add_string("label", on_startup, "");
+	(void)prop->Add_string("path", on_startup, "");
+	(void)prop->Add_string("override_drive", on_startup, "");
+	(void)prop->Add_bool("verbose", on_startup, true);
+
+	return conf;
+}
+
+// Parse a 'Drive' config file and return object with allowed key and value types
+static std::tuple<std::string, std::string, std::string> parse_drive_conf(
```

Comment:
Up to you, I prefer introducing structs above 2-element tuples (pairs), especially in more public APIs (e.g. I tend to only use tuples internally).

Reply:
But probably this is borderline okay 😄

Reply:
Yeah.. this is nasty :laughing: 

I've been thinking about the next round of changes (probably after the release) - and how we can adapt this to also handle image files.

drives/a/floppy-1.img, floppy-2.img, ...}
drives/c/gamedir/{files}
drives/d/{cdrom.bin + .cue}

And I think we'll need a structured return that tells us if we should use `imgmount` or `mount`, and if we'll be mounting the dir itself, ie: `drives/c`, or an image glob pattern, like: `drives/a/*.img` or `drives/d/*.cue`.

Reply:
Ah, yes we definitely need multi image support. But it's a job for later, like you said 😄